src/wallet/wallet.cpp

```diff
@@ -493,6 +496,50 @@ void CWallet::AddToSpends(const uint256& wtxid)
         AddToSpends(txin.prevout, wtxid);
 }
 
+void CWallet::AddToUnspents(const CWalletTx* wtx, bool filterIsMine)
+{
+    uint256 hash = wtx->GetHash();
+
+    for (unsigned int i = 0; i < wtx->vout.size(); ++i)
+    {
+        const COutPoint outpoint(hash, i);
+        pair<TxSpends::const_iterator, TxSpends::const_iterator> range;
+        range = mapTxSpends.equal_range(outpoint);
+
+        // if this output wasn't already marked as spent, we add it as a new UTXO
+        if(range.first == range.second) {
+            isminetype ismine = IsMine(wtx->vout[i]);
+            if(!filterIsMine || ismine) {
+                mapTxUnspents[outpoint] = std::make_pair(wtx, ismine);
+            }
+        }
+    }
+}
+
+void CWallet::RemoveFromUnspents(const COutPoint &outpoint)
+{
+    UnspentsMap::const_iterator uit = mapTxUnspents.find(outpoint);
+    if(uit != mapTxUnspents.end())
+        mapTxUnspents.erase(uit);
+}
+
+void CWallet::RefreshUnspents()
+{
+    for (UnspentsMap::iterator uit = mapTxUnspents.begin(); uit != mapTxUnspents.end();)
+    {
+        const COutPoint &outpoint = uit->first;
+        const CWalletTx* wtx = (uit->second).first;
+
+        isminetype isMine = IsMine(wtx->vout[outpoint.n]);
+        uit->second.second = isMine;
```

Comment:
Don't think that would be very precise. In this case, `IsMine` doesn't return a bool, but a `isminetype` and that value is used later. But I agree with the rest, makes sense to invert the order of the condition and operations.